app/javascript/src/components/Dashboard/index.jsx

```diff
@@ -1,24 +1,47 @@
-import React from "react";
+import React, { useState } from "react";
 
 import { Route, Redirect, Switch } from "react-router-dom";
 
+import { DUMMY_CONTACTS, DUMMY_NOTES, TAGS, ROLES } from "common/dummy";
 import Sidebar from "components/Common/Sidebar";
 import {
   DASHBOARD_ROUTES,
   NOTES_PATH,
   DASHBOARD_PATH,
 } from "components/routeConstants";
 
-const Dashboard = () => (
-  <div className="flex h-screen w-full">
-    <Sidebar />
-    <Switch>
-      {DASHBOARD_ROUTES.map(({ path, component }) => (
-        <Route exact component={component} key={path} path={path} />
-      ))}
-      <Redirect from={DASHBOARD_PATH} to={NOTES_PATH} />
-    </Switch>
-  </div>
-);
+const DashboardContext = React.createContext("dashboard");
 
-export default Dashboard;
+const Dashboard = () => {
+  const [notes, setNotes] = useState(DUMMY_NOTES);
+  const [contacts, setContacts] = useState(DUMMY_CONTACTS);
+  const [tags, setTags] = useState(TAGS);
+  const [roles, setRoles] = useState(ROLES);
+
+  return (
+    <div className="flex h-screen w-full">
+      <Sidebar />
+      <Switch>
+        <DashboardContext.Provider
+          value={{
+            notes,
+            contacts,
+            tags,
+            roles,
+            setNotes,
+            setContacts,
+            setTags,
+            setRoles,
+          }}
+        >
+          {DASHBOARD_ROUTES.map(({ path, component }) => (
+            <Route exact component={component} key={path} path={path} />
+          ))}
+          <Redirect from={DASHBOARD_PATH} to={NOTES_PATH} />
+        </DashboardContext.Provider>
+      </Switch>
+    </div>
+  );
+};
+
+export { Dashboard as default, DashboardContext };
```

Comment:
If you are actually needing to use contexts, then keep similar to how it's done in wheel with a reducer.

Refer:
- https://github.com/bigbinary/wheel/tree/master/app/javascript/src/reducers
- https://github.com/bigbinary/wheel/tree/master/app/javascript/src/contexts

Reply:
Ok. But I might remove the Context API in the fix.